### PR TITLE
Merge sectorbuilder into sectorstorage

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

File: storage/sectorstorage/ffiwrapper/basicfs/fs.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+type Provider struct {
+	Root string
+
+	lk         sync.Mutex
+	waitSector map[sectorFile]chan struct{}
+}
+
+func (b *Provider) AcquireSector(ctx context.Context, id abi.SectorID, existing stores.SectorFileType, allocate stores.SectorFileType, sealing bool) (stores.SectorPaths, func(), error) {
+	os.Mkdir(filepath.Join(b.Root, stores.FTUnsealed.String()), 0755)
```
Comment (whyrusleeping) [inline storage/sectorstorage/ffiwrapper/basicfs/fs.go]: I would prefer we check these errors, ignore 'already exists' errors, but permissions and other fun things should get floated back up

File: storage/sectorstorage/ffiwrapper/basicfs/fs.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	os.Mkdir(filepath.Join(b.Root, stores.FTSealed.String()), 0755)
+	os.Mkdir(filepath.Join(b.Root, stores.FTCache.String()), 0755)
+
+	done := func() {}
+
+	for i := 0; i < 3; i++ {
+		if (existing|allocate)&(1<<i) == 0 {
```
Comment (whyrusleeping) [inline storage/sectorstorage/ffiwrapper/basicfs/fs.go]: wat

File: storage/sectorstorage/ffiwrapper/basicfs/fs.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		prevDone := done
+		done = func() {
+			prevDone()
+			<-ch
+		}
+	}
+
+	return stores.SectorPaths{
+		Id:       id,
+		Unsealed: filepath.Join(b.Root, stores.FTUnsealed.String(), fmt.Sprintf("s-t0%d-%d", id.Miner, id.Number)),
```
Comment (whyrusleeping) [inline storage/sectorstorage/ffiwrapper/basicfs/fs.go]: this join call could be pulled into a variable and reused

File: storage/sectorstorage/ffiwrapper/config.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	x, err := p.RegisteredPoStProof()
+	if err != nil {
+		return 0, err
+	}
+
+	// values taken from https://github.com/filecoin-project/rust-fil-proofs/blob/master/filecoin-proofs/src/constants.rs#L11
+
+	switch x {
+	case abi.RegisteredProof_StackedDRG32GiBPoSt:
+		return 1 << 35, nil
```
Comment (whyrusleeping) [inline storage/sectorstorage/ffiwrapper/config.go]: can we normalize these to 32 << 30, 2 << 10, etc?

Comment (whyrusleeping) [reply storage/sectorstorage/ffiwrapper/config.go]: also this method and the one below are duplicated, no?

File: storage/sectorstorage/ffiwrapper/basicfs/fs.go (hunk 2 above)
Comment (Kubuxu) [reply storage/sectorstorage/ffiwrapper/basicfs/fs.go]: wat wat

**Pull request merged**